src/dmd/func.d

```diff
@@ -2177,6 +2177,76 @@ extern (C++) class FuncDeclaration : Declaration
         return sf;
     }
 
+    /****************************************************
+     * Merge into this function the 'in' contracts of all it overrides.
+     */
+    extern (D) final Statement mergeFrequireInclusivePreview(Statement sf, Expressions* params)
+    {
+        /* If a base function and its override both have an IN contract, then
+         * the override in contract must widen the guarantee of the base contract.
+         * This is checked by generating:
+         *
+         * void derived.in() {
+         *  try {
+         *    ... body of derived.in() ...
+         *  }
+         *  catch () {
+         *    // derived in rejected this argument. so parent must also reject it, or we've tightened the contract.
+         *    base.in();
+         *    assert(false, "Logic error: " ~ thr.msg);
+         *  }
+         */
+
+        foreach (fdv; foverrides)
+        {
+            /* The semantic pass on the contracts of the overridden functions must
+             * be completed before code generation occurs.
+             * https://issues.dlang.org/show_bug.cgi?id=3602
+             */
+            if (fdv.frequires && fdv.semanticRun != PASS.semantic3done)
+            {
+                assert(fdv._scope);
+                Scope* sc = fdv._scope.push();
+                sc.stc &= ~STC.override_;
+                fdv.semantic3(sc);
+                sc.pop();
+            }
+
+            sf = fdv.mergeFrequireInclusivePreview(sf, params);
+            if (sf && fdv.fdrequire)
```

Comment:
please write this as `if (!sf || !fdv.fdrequire) return null;`

Reply:
done